contracts/Gate.sol

```diff
@@ -366,20 +362,20 @@ contract Gate is ProtocolVersioned, Owned {
         // check if the stake request was accepted
         require(stakeRequest.hashLock != bytes32(0));
 
-        bytes32 uuidR = bytes32(0);
         uint256 amountST = uint256(0);
         address stakerAddress = address(0);
-        (uuidR, amountST, stakerAddress) = OpenSTValueInterface(openSTProtocol).revertStaking(_stakingIntentHash);
+        (, amountST, stakerAddress) = OpenSTValueInterface(openSTProtocol).revertStaking(_stakingIntentHash);
 
-        // check if the stake address is not 0
-        require(stakerAddress != address(0));
+        // check if the stake address is correct
+        require(stakerAddress == staker);
```

Comment:
It should be the case that this has to be true, so does this compile with an `assert`?

Reply:
Done !